### DISCUSSION TITLE
Building and ssh tools

### DISCUSSION THREAD

**Discussion by Cronvs** [kanidm/kanidm#2957] [Q&A] [answered]
Greetings, I am very impressed with the project and in the process of integrating it into my homelab.
I have tried to read through the relevant documentation in the book.
Still I have two questions.

Are there anything that one should think about when building without container, any tests or preconfiguration that needs to be done, or do I just use the release/kanidm release/kanidmd targets?

Secondly, I noticed the kanidm-ssh target, I tried finding what the ssh tools are but couldn't find it in the book.
What are the ssh tools that kanidm have?

**Comment by yaleman** [comment]
> Are there anything that one should think about when building without container, any tests or preconfiguration that needs to be done, or do I just use the release/kanidm release/kanidmd targets?

Nearly all our development's done with non-container builds, it runs most places just fine; it's just much harder for our very tiny team to support when folks aren't running a "simple" single state, that's why we suggest folks run containerised deployments.

> kanidm-ssh

Key distribution, which is part of the kanidm-unixd package normally. See https://kanidm.github.io/kanidm/master/integrations/ssh_key_distribution.html?highlight=ssh#server-configuration

**Reply by Cronvs** [reply]
Thank you for your extremely rapid response. Surprised even, since as you say you are a small team.

> Nearly all our development's done with non-container builds, it runs most places just fine; it's just much harder for our very tiny team to support when folks aren't running a "simple" single state, that's why we suggest folks run containerised deployments.

I understand completely. I do not want to miss any of the steps in your normal workflow, of course, to minimize the issues ahead, but at least be at a point where the issues are self-inflicted. Will debug alone either way, since I do not want to waste your time and resources which could be spent on better things.

> Key distribution, which is part of the kanidm-unixd package normally. See https://kanidm.github.io/kanidm/master/integrations/ssh_key_distribution.html?highlight=ssh#server-configuration

Cheers!

**Reply by yaleman** [reply, marked as answer]
It's a pretty simple build, just make sure you set KANIDM_BUILD_PROFILE because it sets where the static web packages will be amongst other things.

**Reply by yaleman** [reply]
And check if your OS is already supported - there's a lot of community-supported packages in various distributions 😄

**Reply by Firstyear** [reply]
Which distro are you trying to run on? It may already be in the supported list :)

**Reply by Cronvs** [reply]
I believe it already is, I am planning on building and running it on Ubuntu/Debian. Still, I like to squeeze out a bit of performance with some settings and native builds.

**Reply by Firstyear** [reply]
We already do x86_64_v3 by default, so you should be all good re performance anyway :) 

Still let us know how you go, but you shouldn't run in to too many issues. If you want to help improve our debian/ubuntu packaging in general that'd be amazing.
